modules/nf-core/modules/deeptools/bamcoverage/main.nf

```diff
@@ -8,28 +8,30 @@ process DEEPTOOLS_BAMCOVERAGE {
         'quay.io/biocontainers/deeptools:3.5.1--py_0' }"
 
     input:
-    tuple val(meta), path(bam), path(bai)
+    tuple val(meta), path(input), path(input_index)
 
     output:
-    tuple val(meta), path("*.bigWig"), emit: bigwig
-    path "versions.yml"              , emit: versions
+    tuple val(meta), path("*.bigWig")   , emit: bigwig, optional: true
+    tuple val(meta), path("*.bedgraph") , emit: bedgraph, optional: true
+    path "versions.yml"                 , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
+    def prefix = task.ext.prefix ?: "${meta.id}.bigWig"
```

Comment:
I don't think the `.bigWig` should be here, I think it was better with it on the script part.
But this is probably an issue to report in the modules repo

Reply:
hm this is the default file ending I choose. apaprently the tool infers whether it is a bedgraph or bigwig file based on file ending provided. I iwll check again if there is a default the tool sets itself

Reply:
Any case, this issue should be resolved on the modules repo, not here, but I can make an issue there so we don't forget if you'd like

Reply:
but then given your answer, and how you already wrote the modules, I would definitively remove the `.bigWig` part from the whole module and only have that as a prefix and test with both bedgraph and bigwig

Reply:
ok sure, my thought there was that we have other modules where a prefix is not mandatory. But am happy to have it mandatory here then. not a hill i am going to die on 😆

Reply:
oh I see, then you would want that as an input?

Reply:
maybe definitively we should start a PR for this modules and discuss it with more people.
I'd be happy to die on another hill that this one too

Reply:
sure sounds good. It would just need to be set in the `modules.config` under `prefix` always and thats it. The documentation is not super clear though on what a default value would be. There is also another parameter that can be used to set the output file format. Maybe there could be weird effects, like setting `--outputfileformat` in `args` but then not having the corresponding ending in `prefix` 🤔 

ok...yes the more I write about it, the more I am sure it could use some more thought